kyaml/kio/filters/container.go

```diff
@@ -44,8 +46,25 @@ type ContainerFilter struct {
 	checkInput func(string)
 }
 
-func (c *ContainerFilter) SetMountPath(path string) {
-	c.mountPath = path
+// StorageMount represents a container's mounted storage option(s)
+type StorageMount struct {
+	// Type of mount e.g. bind mount, local volume, etc.
+	mountType string
+
+	// Source for the storage to be mounted.
+	// For named volumes, this is the name of the volume.
+	// For anonymous volumes, this field is omitted (empty string).
+	// For bind mounts, this is the path to the file or directory on the host.
+	src string
+
+	// The path where the file or directory is mounted in the container.
+	dstPath string
+}
+
+// AddStorageMount adds a mounted storage option to the Container
+func (c *ContainerFilter) AddStorageMount(mountType, src, dstPath string) {
```

Comment:
Instead of making this a function, just make the fields on StorageMount public so they can be directly modified.